rca/account_management/views.py

```diff
@@ -0,0 +1,173 @@
+                "a Parent Student Index Page is required.",
+            )
+            logger.info(
+                "There is not parent Student Index page to create a Student Page under, "
+                f"so one has not been created for user: {student_user}"
+            )
+            return data
+
+        # If a student page already exist for this user, we don't want to create one.
+        student_page = StudentPage.objects.filter(
+            first_name=form.cleaned_data["first_name"],
+            last_name=form.cleaned_data["last_name"],
+        ).exists()
+
+        if student_page:
+            messages.success(
+                self.request,
+                f"The Student account for {student_user} has been created.",
+            )
+            messages.warning(
+                self.request,
+                f"Failed to create a Student Page matching for the new user: {student_user}, "
+                "a Student Page already already exists",
+            )
+            logger.info(
+                "A Student User has been created, but a StudentPage matching "
+                f"the users name already exists: {student_page}, so one has not been created"
+            )
+        else:
+            student_index.add_child(
+                instance=StudentPage(
+                    first_name=form.cleaned_data["first_name"],
+                    last_name=form.cleaned_data["last_name"],
+                    title=f"{form.cleaned_data['first_name']} {form.cleaned_data['last_name']}",
+                    student_user_account=student_user,
+                    live=False,
+                )
+            )
+
+            # Generate a password reset link to be emailed to the user.
+            password_reset_url = self.request.build_absolute_uri(
+                get_set_password_url(student_user)
+            )
+            # Notify the user of the account.
+            email_subject = _("Your Student account has been created")
+            email_body = render_to_string(
+                "account_management/admin/emails/notify_user_on_creation.txt",
+                {
+                    "finish_registration_url": password_reset_url,
+                    "user": student_user,
+                    "PASSWORD_RESET_TIMEOUT_DAYS": settings.PASSWORD_RESET_TIMEOUT_DAYS,
+                },
+            )
+
+            send_mail(
+                email_subject,
+                email_body,
+                "do-not-reply@rca.ac.uk",
+                [student_user.email],
+            )
+            messages.success(
+                self.request,
+                f"The Student Page for {student_user} has been created."
+                f"A Notification email has been sent to {student_user.email}",
+            )
+
+        return data
+
+
+class CustomLoginView(LoginView):
+    """Custom login view to redirect students to their profile page
+    """
+
```

Comment:
AC: As a student after resetting my password, then I can log in and I am redirected to my profile in Wagtail CMS